libparsec/crates/platform_storage/src/web/db.rs

```diff
@@ -0,0 +1,126 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use indexed_db_futures::{web_sys::IdbTransactionMode, IdbDatabase, IdbQuerySource};
+use libparsec_types::anyhow;
+use serde::{de::DeserializeOwned, Serialize};
+
+pub(super) async fn get_value<K, V>(
+    conn: &IdbDatabase,
+    store: &str,
+    index: &str,
+    key: K,
+) -> anyhow::Result<Option<V>>
+where
+    K: Serialize,
+    V: DeserializeOwned,
+{
+    let tx = conn
+        .transaction_on_one(store)
+        .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+
+    let store = tx
+        .object_store(store)
+        .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+
+    let indexed_store = store.index(index).map_err(|e| anyhow::anyhow!("{e:?}"))?;
```

Comment:
Would it be possible to factorize those steps shared with `get_values` ?

Reply:
Yes

Reply:
I think this is not possible without macro due to borrowed data :(

Reply:
The borrowed data is `store` that does not live long enough ? a macro could be used since it would be small

Reply:
Yes

Reply:
If it's too hard we can skip the factorization :sweat_smile: